core/src/main/java/hudson/model/Queue.java

```diff
@@ -2620,6 +2620,14 @@ void setCauseOfBlockage(CauseOfBlockage causeOfBlockage) {
             this.causeOfBlockage = causeOfBlockage;
         }
 
+        public boolean isCauseOfBlockageNull() {
+            if (causeOfBlockage == null) {
+                return true;
+            } else {
+                return false;
+            }
+        }
```

Comment:
The fix itself looks good. Thanks for the quick response! On a procedural note, a new public API either needs to be documented or restricted, as described in the **Submitter checklist.** In this case the only consumer is in core and I do not think we anticipate additional external consumers, so I would lean toward restricting the new API with `@Restricted(NoExternalUse.class)`.

Reply:
Done